### ISSUE TITLE
Configuration settings for type Bool are not saved

### ISSUE THREAD

**Issue by servusoft** [arangodb/arangodb#2067]
## my environment running ArangoDB

I'm using the latest ArangoDB of the respective release series:
- [x] 2.8
- [x] 3.0
- [x] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [x] Linux 
  - [x] Debian .deb
  - [x] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:
### this is an AQL-related issue:

[ ] I'm using graph features

I'm issuing AQL via:
- [x] web interface with this browser:     running on this OS
- [ ] arangosh
- [ ] this Driver:

I've run `db._explain("<my aql query>")` and it didn't shed more light on this.
The AQL query in question is:

The issue can be reproduced using this dataset:

Please provide a way to create the dataset to run the above query on; either by a gist with an arangodump, or `db.collection.save({my: "values"}) statements. If it can be reproduced with one of the ArangoDB example datasets, it's a plus.
### this is a web interface-related issue:

I'm using the web interface with this browser:     running on this OS:
- [x] authentication is enabled?
- [ ] using the cluster?
- [x] _system database?

These are the steps to reproduce:
1) open the browser on http://127.0.0.1:8529
2) log in as ...
3) use database [ ] `_system` [ ] other: 
4) click to ...
...

The following problem occurs: [Screenshot?] 
Instead I would be expecting: 
### this is an installation-related issue:

Describe which steps you carried out, what you expected to happen and what actually happened.
1. Create a service SERVICES->Add service->New service ...
2. Edit "manifest.json": 
   Append config-part:
   
   "configuration": { 
     "configBool": { 
       "description": "Testing boolean setting", 
       "type": "boolean",
       "default" : "false"
     } 
   },
3. Save "manifest.json" 

there is follow bugs:
1. After changing of "manifest.json" the setting in front end  will be not updated
   - after reload of services 
   - changing to "Development"/ "Production"mode has no effect
   - Is is possable to update only as download & replace app
     It be needed a option "Reload app" or "Restart services"
2. After reload of app (services) 
   - default-value  "false" (from manifest.json)  is not set
   - changeging of value  (in Frontend)  is not saved and show allwais "on"
3. After removing of a config-parts from manifest.json shown follow warning under LOGS in frontend:
   -<i>  Stored configuration for app "/test" has errors: </i>
   -<i>  Unexpected option "configBool" </i>
   Probaly needed an option "Remove config-item"

**Comment by Simran-B** [comment]
Might not be related, but shouldn't the default be `"default": false`? It is a string in your example.

**Comment by servusoft** [comment]
You're right, default value false (without "") worked! Thus, a bug less;)
But saving the setting is still not working :(

**Comment by pluma** [comment]
This should be fixed in the current devel, 3.1 and 3.0 branches.

Changes to the configuration were saved using UPDATE instead of REPLACE, leading to weird merging behaviour in some cases. This was definitely a bug.

@servusoft can you try the GitHub version or the next bugfix/minor release when it comes out?

**Comment by servusoft** [comment]
@pluma, I have tested on 3.0, 3.1-rc3 and 3.1 devel but it works not pretty:
I add follow part  to manifest.json
  "configuration": {
    "cfgBool": {
      "description": "Test Boolean",
      "type": "boolean"
    }
  }

If I save cfgBool als setted *on* (with *apply* button)
it is setted in req.context:.configuration  as *on*:
configuration: {
  cfgBool: true
},
but if I (re)open the service again, it's shown as *off* and can't see it's on or off

**Comment by pluma** [comment]
This seems to be an issue with the web interface, not Foxx itself. @hkernbach can you check the rendering logic for this?

**Comment by servusoft** [comment]
I think the bug is in the `createCheckboxEntry` method from `modalView.js`.

tab `Setting` in aardvark call `showConfigDialog` from `applicationDetailView.js`:

```
showConfigDialog: function () {
...
  if (obj.type === 'boolean' || obj.type === 'bool') {
    methodName = 'createCheckboxEntry';
    obj.default = obj.default || false;
    defaultValue = obj.default || false;
    currentValue = obj.current || false;
  }
...
 
return window.modalView[methodName](
  'app_config_' + name,
   name,
   currentValue,
   obj.description,
   defaultValue,
   mandatory,
   checks
);
```
over parameter `checked` will passed the `defaultValue`. 
but the saved value (currentValue) has no effect:
 
`modalView.js`:
```
createCheckboxEntry: function (id, label, value, info, checked) {
  var obj = createTextStub(this.tables.CHECKBOX, label, value, info);
  obj.id = id;
  if (checked) {
    obj.checked = checked;
  }
  return obj;
}
```
the method createCheckboxEntry in modalView.js should be probely extended as 
```
...
  if (value) {
    obj.checked = value;
  }
...  
```

note: the app should be recompiled, because aardvark uses the compiled version (app.min.js)

**Comment by hkernbach** [comment]
Thanks, added fix to 3.0, 3.1 and devel.
